CoreFunctionality/src/au/gov/asd/tac/constellation/functionality/startup/ConstellationLAFSettings.java

```diff
@@ -0,0 +1,333 @@
+ *
+ * @author OrionsGuardian
+ */
+public class ConstellationLAFSettings {
+
+    public static void applyTabColorSettings() {
+        JFrame mainframe = (JFrame) WindowManager.getDefault().getMainWindow();
+        String currentLafName = UIManager.getLookAndFeel().getName().toUpperCase();
+        if (currentLafName.startsWith("WINDOWS")) {
+            initWindowsTabColors();
+        } else if (currentLafName.contains("NIMBUS")) {
+            initNimbusTabColors(currentLafName.contains("DARK"));
+        } else if (currentLafName.contains("FLAT")) {
+            initFlatLafTabColors(currentLafName.contains("DARK"));
+        } else if (currentLafName.contains("METAL")) {
+            initMetalTabColors(currentLafName.contains("DARK"));
+        }
+        SwingUtilities.updateComponentTreeUI(mainframe);
+    }
+
+    /**
+     * Applies a Blue color theme for the main window tabs when using either the
+     * <b>Windows</b> or <b>Windows Classic</b> Look and Feel.
+     */
+    private static void initWindowsTabColors() {
+        Color selectedUpperLightBlue = new Color(225, 235, 255);
+        Color activeMidSectionBlue = new Color(140, 185, 255);
+        Color selectedLowerDarkBlue = new Color(50, 130, 255);
+        Color unselectedUpperGray = new Color(220, 220, 220);
+        Color unselectedLowerGreyBlue = new Color(165, 175, 185);
+
+        UIManager.getDefaults().put("tab_sel_fill", activeMidSectionBlue);
+        UIManager.getDefaults().put("tab_focus_fill_upper", selectedUpperLightBlue);
+        UIManager.getDefaults().put("tab_focus_fill_lower", selectedLowerDarkBlue);
+        UIManager.getDefaults().put("tab_unsel_fill_upper", unselectedUpperGray);
+        UIManager.getDefaults().put("tab_unsel_fill_lower", unselectedLowerGreyBlue);
+        UIManager.getDefaults().put("tab_mouse_over_fill_upper", selectedUpperLightBlue);
+        UIManager.getDefaults().put("tab_mouse_over_fill_lower", activeMidSectionBlue);
+        UIManager.getDefaults().put("tab_attention_fill_upper", activeMidSectionBlue);
+        UIManager.getDefaults().put("tab_attention_fill_lower", selectedLowerDarkBlue);
+
+        try {
+            Class<?> tabDisplayer = ((Class<?>) UIManager.getDefaults()
+                    .get("org.netbeans.swing.tabcontrol.plaf.Windows8VectorViewTabDisplayerUI")).getSuperclass();
+            
+            // reset static field "colorsReady" value to false
+            Field colsReady = tabDisplayer.getDeclaredField("colorsReady");
+            colsReady.setAccessible(true);
+            colsReady.setBoolean(tabDisplayer, false);
+        } catch (IllegalAccessException | IllegalArgumentException | 
+                NoSuchFieldException | SecurityException e) {
+            String errorMessage = " >>> Error applying Windows LaF colors : " + e.toString();
+            Logger.getLogger(ConstellationLAFSettings.class.getName()).info(errorMessage);
```

Comment:
are you able to extract the logger into a private constant (similar to elsewhere in the codebase)?

Reply:
Done